activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -130,27 +130,37 @@ def initialize(*addresses)
       end
 
       # Increment a cached value. This method uses the memcached incr atomic
-      # operator and can only be used on values written with the +:raw+ option.
-      # Calling it on a value not stored with +:raw+ will initialize that value
-      # to zero.
+      # operator and can only be used on values written with the +:raw+ option
+      # or initially set using a previous increment call.
+      # Calling it on a value not stored with :raw will fail and return nil.
+      # :initial option can be used for initializing the value to other than zero, if no
+      # initial option is set, the method will assume amount as intial value
```

Comment:
Here's an attempt to document all of this method's behaviour:

```ruby
      # Increment a cached value. This method uses the memcached incr atomic
      # operator and can only be used on values written with the +:raw+ option.
      # Calling it on a value not stored with +:raw+ will fail and return +nil+.
      #
      # Calling it on a non-existent key will set the value to +amount+, or to the
      # value of the +:initial+ option if it's set. If +:initial+ is set to +nil+,
      # calling #increment on a non-existent value will fail and return +nil+.
      #
      #   cache.increment("foo")               # => "1"
      #   cache.increment("bar", initial: 2)   # => "2"
      #   cache.increment("baz", initial: nil) # => nil
```

I'm still not totally happy with it though, it's a collection of facts rather than a guide to using the method. I'm also not happy with documenting the fact that raw integers are returned as strings, since we want to change that too.

Reply:
How about this:
```
# Increment a cached value. This method uses the memcached incr atomic
# operator and can be used in one of two ways. First way is to increment
# an already stored value, which has to be written using the +:raw: option.
# Second way is to let the method initialise the value itself either by setting
# the +:initial: option or using the +amount+ as default:

#   cache.write("baz", 0, raw: true); cache.increment("baz") # => "1"
#   cache.increment("foo")                                   # => "1"
#   cache.increment("bar", initial: 100)                     # => "100"

# Calling the method on a value not stored with +:raw+ will fail and return +nil+.
# If +:initial+ is set to +nil+, calling the method on a non-existent value will 
# also fail and return +nil+.
```

I am OK keeping strings as returns, we can update that when we change the return type to integers